### PR TITLE
Improve compiler flags

### PR DESCRIPTION
**Pull request by danholdaway**
- ecbuild default flags
- add option to overload r4 for building fv3 in single precision

Possible this will change results for mom6 so please test.

### CODE REVIEW

**Comment by danholdaway** [comment]
@guillaumevernieres @travissluka this is a pretty temporary thing since we can use the cmake version of fms coming soon. Then we'll add fms to the jedi-stack and modules and we'll have r4 and r8 libraries available simultaneously. But can you just quickly check this branch doesn't break soca?

File: cmake/compiler_flags_Intel_Fortran.cmake [hunk 1]
```diff
@@ -1,41 +0,0 @@
-# (C) Copyright 2018 UCAR.
-#
-# This software is licensed under the terms of the Apache Licence Version 2.0
-# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
-
-####################################################################
-# FLAGS COMMON TO ALL BUILD TYPES
-####################################################################
-
-set( CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fno-alias -auto -safe-cray-ptr -ftz -assume byterecl -i4 -r8 -nowarn -sox -traceback" )
```
Comment (travissluka) [inline cmake/compiler_flags_Intel_Fortran.cmake:10]: I'm getting segfaults in MOM when compiling in intel now, gcc seems to be fine. I suspect it needs some of these flags added back but haven't checked yet

**Comment by travissluka** [comment]
I've tracked down the segfaults in MOM to the removal of the `CMAKE_Fortran_FLAGS_<BUILD_TYPE>` overrides here. By default I'm showing the following are set for the entire bundle when using intel19:
image
The `-heap-arrays 32` flag seems to be the specific one causing problems. Removing that works for me.
@guillaumevernieres I'd be curious to see if it crashes for you when compiled in intel19/release. Im not sure what the best way is to fix this without having FMS clear out all the `CMAKE_Fortran_FLAGS_<BUILD_TYPE>`  variables

**Comment by danholdaway** [comment]
Thanks for looking into it. I was just testing on Discover and finding it still crashed when I added those extra defs back in. So I guess we need to keep the intel toolchain. I don't mind doing that, hopefully be can get rid of fms from the bundle fairly soon anyway.

**Comment by danholdaway** [comment]
@travissluka I just pushed a change that has the soca tests passing on discover. Can you test again to make sure it works for you?

**Comment by danholdaway** [comment]
And it doesn't change any of your test output so hopefully good to merge this once you give the sign off.

**Comment by mathomp4** [comment]
Came upon this in my GitHub Notifications. Indeed, I think I have/had to remove `-heap-arrays 32` from MOM5 compilation in GEOS. Interestingly, our MOM6 build doesn't seem to care (as I just looked and we are using it there). 

Well, yet.

**Comment by danholdaway** [comment]
Thanks for letting us know @mathomp4. Perhaps there are differences in the way the grid gets generated. That seems to be where it's failing.

**Comment by danholdaway** [comment]
Thanks @travissluka

**Pull request merged**